### PR TITLE
Constants Koans

### PR DESCRIPTION
**Pull request by napalm684**
Koans related to C# constants.

### CODE REVIEW

File: Koans/AboutConstants.cs [hunk 1]
```diff
@@ -0,0 +1,114 @@
+using DotNetCoreKoans.Engine;
+using Xunit;
+
+namespace DotNetCoreKoans.Koans
+{
+    public class AboutConstants : Koan
+    {
+        const string C = "nested";
```
Comment (NotMyself) [inline Koans/AboutConstants.cs:8]: I do not think this example is relevant to C#. The ruby koans demonstrates that constants can be declared globally or in a class definition. Global constants are not really a thing in C#.

File: Koans/AboutConstants.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+        [Step(1)]
+        public void ConstantsMustBeInitalizedAsDeclared()
```
Comment (NotMyself) [inline Koans/AboutConstants.cs:11]: This is a good first step. I think the next step should demonstrate that a constant cannot be changed.

File: Koans/AboutConstants.cs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            const int days = 365;
+
+            const double daysPerWeek = (double)days / (double)weeks;
+            const double daysPerMonth = (double)days / (double)months;
+            Assert.Equal(FILL_ME_IN, daysPerWeek);
+            Assert.Equal(FILL_ME_IN, daysPerMonth);
+        }
+
+        [Step(4)]
+        public void NestedConstantsAreReferencedWithRelativePaths()
```
Comment (NotMyself) [inline Koans/AboutConstants.cs:40]: This step is related to the first comment. I don't think it has meaning in C#.

File: Koans/AboutConstants.cs [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        {
+            Assert.Equal("nested", C);
+        }
+
+        [Step(5)]
+        public void NestedConstantsAreReferencedByTheirCompletePath()
```
Comment (NotMyself) [inline Koans/AboutConstants.cs:46]: same

File: Koans/AboutConstants.cs [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+            {
+                public int LegsInNestedAnimal()
+                {
+                    return Legs;
+                }
+            }
+        }
+
+        [Step(6)]
+        public void NestedClassesInheritConstantsFromEnclosingClasses()
```
Comment (NotMyself) [inline Koans/AboutConstants.cs:70]: This is a much better example

**Review by NotMyself — CHANGES_REQUESTED** [review]
Overall this is a great start. I left a few comments. Also it would be nice to ad some comments that explain a bit what is going on. Take a look at the other koans for examples.

File: Koans/AboutConstants.cs (hunk 2 above)
Comment (napalm684) [reply Koans/AboutConstants.cs:11]: How do you propose to demonstrate that? Changing a constant value is a compilation error?

Comment (napalm684) [reply Koans/AboutConstants.cs:9]: @NotMyself please see latest.

**Review by NotMyself — APPROVED** [review]

**Comment by NotMyself** [comment]
@all-contributors please add @napalm684 for code

**Pull request merged**